clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp

```diff
@@ -633,37 +633,110 @@ static Expected<StringRef> runLLVMToSPIRVTranslation(StringRef InputTable,
+  auto MBOrErr = MemoryBuffer::getFile(File);
+  if (!MBOrErr)
+    return createFileError(File, MBOrErr.getError());
+
+  line_iterator LI(**MBOrErr);
+  // check the header.
+  if (LI.is_at_eof() || *LI != "[Code|Properties|Symbols]")
+    return createStringError(inconvertibleErrorCode(),
+                             "invalid SYCL Table file.");
+
+  ++LI;
+  SmallVector<SYCLImage> Images;
+  for (; !LI.is_at_eof(); ++LI) {
+    StringRef Line = *LI;
+    SmallVector<StringRef, 3> Elems;
+    Line.split(Elems, '|'); // expected elems are Code, Properties and Symbols.
+
+    if (Elems.size() != 3)
+      return createStringError(inconvertibleErrorCode(),
+                               "invalid SYCL Table file.");
+
+    auto ImageOrErr = readFile(Elems[0]);
+    if (!ImageOrErr)
+      return ImageOrErr.takeError();
+
+    auto PropertiesOrErr = readPropertyRegistryFromFile(Elems[1]);
+    if (!PropertiesOrErr)
+      return PropertiesOrErr.takeError();
+
+    auto SymbolsOrErr = readFile(Elems[2]);
+    if (!SymbolsOrErr)
+      return SymbolsOrErr.takeError();
+
+    SYCLImage Image;
+    Image.Image = std::move(*ImageOrErr);
+    Image.PropertyRegistry = std::move(**PropertiesOrErr);
+    Image.Entries = std::move(*SymbolsOrErr);
+    Images.push_back(std::move(Image));
+    // TODO: where are compile-opts, link-opts and other stuff?
+  }
+
+  return Images;
+}
+
+// Run clang-offload-wrapper's functionality
+// TODO: this function should be removed in favor of community flow
+Expected<StringRef> wrapSYCLBinariesFromFile(StringRef InputFile,
+                                             const ArgList &Args) {
+  auto OutputFileOrErr = createOutputFile(sys::path::filename(ExecutableName) + ".sycl.image.wrapper", "bc");
+  if (!OutputFileOrErr)
+    return OutputFileOrErr.takeError();
+
+  StringRef OutputFilePath = *OutputFileOrErr;
+  if (DryRun) {
+    errs() << formatv(" offload-wrapper: input: {0}, output: {1}\n", InputFile, OutputFilePath);
+    return OutputFilePath;
+  }
+
+  auto ImagesOrErr = readImages(InputFile);
+  if (!ImagesOrErr)
+    return ImagesOrErr.takeError();
+
+  auto &Images = *ImagesOrErr;
 
-  SmallVector<StringRef, 8> CmdArgs;
-  CmdArgs.push_back(*ClangOffloadWrapperPath);
-  CmdArgs.push_back(Saver.save("-o=" + *TempFileOrErr));
-  llvm::Triple HostTriple(
-      Args.getLastArgValue(OPT_host_triple_EQ, sys::getDefaultTargetTriple()));
-  CmdArgs.push_back(Saver.save("-host=" + HostTriple.str()));
   const llvm::Triple Triple(Args.getLastArgValue(OPT_triple_EQ));
   SmallString<128> TargetTripleOpt = Triple.getArchName();
-  CmdArgs.push_back(Saver.save("-target=" + TargetTripleOpt));
-  CmdArgs.push_back("-kind=sycl");
-  CmdArgs.push_back("-batch");
-  CmdArgs.push_back(InputFile);
-  if (Error Err = executeCommands(*ClangOffloadWrapperPath, CmdArgs))
-    return std::move(Err);
-  return *TempFileOrErr;
+  LLVMContext C;
+  Module M("offload.wrapper.object", C);
```

Comment:
Can we please create a const string with this value and use the string here?

Reply:
You want something like this?
```cpp
LLVMContext C;
StringRef ModuleName = "offload.wrapper.object";
Module M(ModuleName, C);
```